Code/Geometry/Trajectory.h

```diff
@@ -0,0 +1,143 @@
+    }
+    /*! \brief Frees the pointer to the array of doubles where the
+        coordinates for this Snapshot are stored
+     */
+  private:
+    // Pointer to the parent Trajectory object
+    const Trajectory *d_trajectory;
+    // Energy for this set of coordinates
+    double d_energy;
+    // shared array to Snapshot coordinates
+    boost::shared_array<double> d_pos;
+};
+
+class Trajectory {
+  public:
+    typedef std::vector<boost::shared_ptr<Snapshot> > SnapshotSPtrVect;
+    /*! \brief Constructor
+        \param dimension represents the dimensionality of this Trajectory's coordinate tuples;
+        this is normally 2 (2D coordinates) or 3 (3D coordinates)
+        \param numPoints is the number of coordinate tuples associated to each Snapshot
+     */
+    Trajectory(unsigned int dimension, unsigned int numPoints);
+    /*! \brief Copy constructor
+     */
+    Trajectory(const Trajectory &other);
+    /*! \return the dimensionality of this Trajectory's coordinate tuples
+     */
+    unsigned int dimension() const {
+      return d_dimension;
+    }
+    /*! \return the number of coordinate tuples associated to each Snapshot
+     */
+    unsigned int numPoints() const {
+      return d_numPoints;
+    }
+    /*! \return the number of Snapshots associated to this Trajectory
+     */
+    size_t size() const {
+      return d_snapshotVect.size();
+    }
+    /*! \brief Appends a Snapshot to this Trajectory
+        \param s is the Snapshot to be added; the Trajectory
+               takes ownership of the snapshot coordinates
+        \return the zero-based index position of the added Snapshot
+     */
+    unsigned int addSnapshot(Snapshot *s);
+    /*! \param snapshotNum is the zero-based index of the retrieved Snapshot
+        \return a const reference to the relevant Snapshot in the Trajectory
+     */
+    Snapshot *getSnapshot(unsigned int snapshotNum) const;
+    /*! \brief Inserts a Snapshot into this Trajectory
+        \param snapshotNum is the zero-based index of the Trajectory's Snapshot
+               before which the Snapshot s will be inserted
+        \param s is the Snapshot to be inserted; the Trajectory takes ownership of the Snapshot
+        \return the zero-based index position of the inserted Snapshot
+     */
+    unsigned int insertSnapshot(unsigned int snapshotNum, Snapshot *s);
+    /*! \brief Removes a Snapshot from this Trajectory
+        \param snapshotNum is the zero-based index of Snapshot to be removed
+        \return the zero-based index position of the Snapshot after the
+                removed one; if the last snapsot was removed, it returns the
+                size of the trajectory
+     */
+    unsigned int removeSnapshot(unsigned int snapshotNum);
+    /*! \brief Reads coordinates from an AMBER trajectory file
+               into the Trajectory object
+        \return the number of Snapshot objects read in
+     */
+    unsigned int readAmber(const std::string &fName);
```

Comment:
These should probably not be part of this class, but separate functions.  I usually don't have IO in the data structure, but objects that use the data structure.  This separation of concern is also a good test of the public api.